internal/operands/common-instancetypes/reconcile.go

```diff
@@ -0,0 +1,169 @@
+package common_instancetypes
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+	instancetypeapi "kubevirt.io/api/instancetype"
+	instancetypev1alpha2 "kubevirt.io/api/instancetype/v1alpha2"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Define RBAC rules needed by this operand:
+// +kubebuilder:rbac:groups=instancetype.kubevirt.io,resources=virtualmachineclusterinstancetypes,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=instancetype.kubevirt.io,resources=virtualmachineclusterpreferences,verbs=get;list;watch;create;update;patch;delete
+
+const (
+	operandName                      = "common-instancetypes"
+	operandComponent                 = common.AppComponentTemplating
+	BundleDir                        = "data/common-instancetypes-bundle/"
+	ClusterInstancetypesBundlePrefix = "common-clusterinstancetypes-bundle"
+	ClusterPreferencesBundlePrefix   = "common-clusterpreferences-bundle"
+)
+
+type commonInstancetypes struct {
+	virtualMachineClusterInstancetypes []instancetypev1alpha2.VirtualMachineClusterInstancetype
+	virtualMachineClusterPreferences   []instancetypev1alpha2.VirtualMachineClusterPreference
+}
+
+var _ operands.Operand = &commonInstancetypes{}
+
+func fetchVirtualMachineClusterInstancetypeBundle(bundlePath string) ([]instancetypev1alpha2.VirtualMachineClusterInstancetype, error) {
```

Comment:
Is there a way to deduplicate this and `fetchVirtualMachineClusterPreferencesBundle`? Maybe by using generics?

Reply:
Is Go >=1.18 used everywhere up and downstream now?

Reply:
Yes, 1.19 is used u/s and d/s.